src/infrastructure/api/index.ts

```diff
@@ -0,0 +1,112 @@
+import AWS from 'aws-sdk';
+import express, { Request } from 'express';
+import mysql from 'mysql2/promise';
+import vehicleQueryFunctionFactory from '../../app/vehicleQueryFunctionFactory';
+import testQueryFunctionFactory from '../../app/testQueryFunctionFactory';
+import { getResultsDetails, getVehicleDetails } from '../../domain/enquiryService';
+import ParametersError from '../../errors/ParametersError';
+import ResultsEvent from '../../interfaces/ResultsEvent';
+import VehicleEvent from '../../interfaces/VehicleEvent';
+import DatabaseService from '../databaseService';
+import SecretsManagerService from '../secretsManagerService';
+
+const app = express();
+const router = express.Router();
+
+const { API_VERSION } = process.env;
+
+// Declare middlewares
+/**
+ * bodyParser, error handling, logger, etc..
+ * http://expressjs.com/en/starter/basic-routing.html
+ * http://expressjs.com/en/guide/using-middleware.html
+ */
+
+/**
+ * app level middlewares
+ * app.use('/path', (req, res, next) => {
+ * chain middlewares
+ * next()
+ * })
+ */
+router.use((req, _response, next) => {
+  // TODO Add logger lib like Winston or Morgan
+  console.log('path');
+  console.log(req.path);
+  next();
+});
+
+// Debug router before we start proxying  requests from /v<x> psth
+router.get('/', (_request, res) => {
+  res.send({ ok: true });
+});
+
+router.get('/version', (_request, res) => {
+  res.send({ version: API_VERSION });
+});
+
+router.get(
+  '/vehicle',
+  (
+    request: Request<Record<string, unknown>, string | Record<string, unknown>, Record<string, unknown>, VehicleEvent>,
```

Comment:
did not know the `unknown` type nice, will do some research to see how it differs from `undefined` :)